### PR TITLE
[Feature] Support ViTPose

### PR DESCRIPTION
**Pull request by LareinaM**
## Motivation

Add implementation of ViTPose on MMPose 1.0

## Modification

- Add six config files
- Add layer decay optimizer
- In `HeatmapHead`, add parameter `upsample`, default to zero (no effect on previous codes), `resize` in `BaseHead`

## Checklist

**Before 

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by jin-s13** [comment]
Have you tried training a base ViT model to check the accuracy?

**Comment by LareinaM** [comment]
>

**Comment by LareinaM** [comment]
### Result of the current implementation

> With classic decoder

| Arch | Input Size | AP |  AR |
| :-------------------------------------------- | :--------: | :---: | :-------------: |
| ViTPose-S | 256x192 | 0.732 | 0.788 |
| ViTPose-B | 256x192 | 0.757 | 0.906 | 0.829 | 0.810 |
| ViTPose-L | 256x192 | 0.779 | 0.914 | 0.850 | 0.833 |
| ViTPose-H | 256x192 | 0.772 | 0.910 | 0.844 | 0.828 |

> With simple decoder 

| Arch | Input Size | AP |  AR |
| :-------------------------------------------- | :--------: | :---: | :-------------: |
| ViTPose-B | 256x192 | 0.754 | 0.807 |
| ViTPose-L | 256x192 | 0.780 | 0.833 |
| ViTPose-H | 256x192 | 0.768 | 0.825 |

### Result of original ViTPose implementation

> With classic decoder

| Model | Input Size | AP | AR |
| :----: | :----: | :----: | :----: |
| ViTPose-S | 256x192 | 0.738 | 0.792 | 
| ViTPose-B | 256x192 | 0.758 | 0.811 | 
| ViTPose-L | 256x192 | 0.783 | 0.835 |
| ViTPose-H | 256x192 | 0.791 | 0.841 |

> With simple decoder

| Model | Input Size | AP | AR |
| :----: | :----: | :----: | :----: |
| ViTPose-B | 256x192 | 0.755 | 0.809 | 
| ViTPose-L | 256x192 | 0.782 | 0.834 |
| ViTPose-H | 256x192 | 0.789 | 0.840 |

File: mmpose/models/heads/heatmap_heads/heatmap_head.py [hunk 1]
```diff
@@ -64,6 +64,8 @@ class HeatmapHead(BaseHead):
             keypoint coordinates from the network output. Defaults to ``None``
         init_cfg (Config, optional): Config to control the initialization. See
             :attr:`default_init_cfg` for default settings
+        extra (dict, optional): Extra configurations.
+            Defaults to ``None``
```
Comment (ly015) [inline mmpose/models/heads/heatmap_heads/heatmap_head.py:68]: The argument `extra` is convenient for extending the head class but may confuse users. We should keep a well-defined interface where every argument has a clear meaning and a detailed usage description in the docstring.

Here I think maybe we can use existing arguments `conv_out_channels`, `conv_kernel_sizes`, and `has_final_layer` to configure the final conv layers.

File: mmpose/models/heads/heatmap_heads/heatmap_head.py [hunk 2]
```diff
@@ -101,6 +104,21 @@ def __init__(self,
             self.decoder = KEYPOINT_CODECS.build(decoder)
         else:
             self.decoder = None
+        self.upsample = 0
```
Comment (ly015) [inline mmpose/models/heads/heatmap_heads/heatmap_head.py:107]: I suggest adding a new argument, e.g. `input_upsample` or `input_rescale`.

**Review by ly015 — APPROVED** [review]

**Pull request merged**